compiler/rustc_log/src/lib.rs

```diff
@@ -84,9 +94,13 @@ pub fn init_env_logger(env: &str) -> Result<(), Error> {
         .with_verbose_entry(verbose_entry_exit)
         .with_indent_amount(2);
     #[cfg(parallel_compiler)]
-    let layer = layer.with_thread_ids(true).with_thread_names(true);
+    let log_layer = log_layer.with_thread_ids(true).with_thread_names(true);
+
+    let error_layer = tracing_error::ErrorLayer::default();
 
-    let subscriber = tracing_subscriber::Registry::default().with(filter).with(layer);
+    let subscriber = tracing_subscriber::Registry::default()
+        .with(error_layer.with_filter(error_filter))
```

Comment:
this layer seems to be affecting the log layer. Not sure what's going on.

Reply:
they interact with the same registry in the background, some of the code ended up being shared, so that may be causing it. I can try reversing their order so the log layer gets applied first and see how that affects things.